sdk/messaging/azservicebus/message.go

```diff
@@ -4,105 +4,331 @@
 package azservicebus
 
 import (
+	"context"
+	"fmt"
+	"time"
+
 	"github.com/Azure/azure-amqp-common-go/v3/uuid"
-	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus/internal"
 	"github.com/Azure/go-amqp"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/devigned/tab"
 )
 
-const (
-	annotationPartitionKey    = "x-opt-partition-key"
-	annotationViaPartitionKey = "x-opt-via-partition-key"
+type (
+	// ReceivedMessage is a received message from a Client.NewReceiver() or Client.NewProcessor().
+	ReceivedMessage struct {
+		Message
+
+		LockToken              *amqp.UUID
```

Comment:
This exchange type, and potentially UUIDs in general, needs to be revisited.

Reply:
To clarify - do you mean in this PR? Or do we already have a conversation happening?

Reply:
How do callers use the `LockToken`, can we simply expose it as a `[16]byte`?

For some context.  In track 1 we exposed UUIDs using a thrid-party UUID type which was problematic for various reasons.  The conclusion we arrived at (for HTTP-based services) was to expose UUIDs in their native format (strings) so that callers can use their UUID-parsing package of choice.

Reply:
This would also imply that I need to remove the `amqp.UUID` type.

Reply:
Took your suggestion, switch it to [16]byte array.